test/test_types.js

```diff
@@ -2130,6 +2130,52 @@ suite('types', function () {
       var E = t.getRecordConstructor();
       var err = new E('MyError');
       assert(err instanceof Error);
+      if (typeof Error.captureStackTrace === 'function') {
+        assert(err.stack, 'Error.captureStackTrace was not called');
+      }
+    });
+
+    test('error type - stack is not overridden', function() {
```

Comment:
We could if errors end up differing significantly from records; for now I'd say let's stick with the current hierarchy.